static/app/components/tabs/tab.tsx

```diff
@@ -0,0 +1,178 @@
+import {forwardRef} from 'react';
+import styled from '@emotion/styled';
+import {useTab} from '@react-aria/tabs';
+import {mergeProps, useObjectRef} from '@react-aria/utils';
+import {TabListState} from '@react-stately/tabs';
+import {Node, Orientation} from '@react-types/shared';
+
+import space from 'sentry/styles/space';
+
+interface TabProps<T> {
+  item: Node<T>;
+  orientation: Orientation;
+  overflowing: boolean;
+  state: TabListState<T>;
+}
+
+/**
+ * Renders a single tab item. This should not be imported directly into any
+ * page/view – it's only meant to be used by <TabsList />. See the correct
+ * usage in tabs.stories.js
+ */
+function BaseTab<T>(
+  {item, state, orientation, overflowing}: TabProps<T>,
+  forwardedRef: React.ForwardedRef<HTMLLIElement>
+) {
+  const ref = useObjectRef(forwardedRef);
+
+  const {key, rendered} = item;
+  const {tabProps, isSelected, isDisabled} = useTab({key}, state, ref);
+
+  return (
+    <TabWrap
+      {...mergeProps(tabProps)}
+      disabled={isDisabled}
+      selected={isSelected}
+      overflowing={overflowing}
+      orientation={orientation}
+      ref={ref}
+    >
+      <HoverLayer orientation={orientation} />
+      <FocusLayer orientation={orientation} />
+      {rendered}
+      <TabSelectionIndicator orientation={orientation} selected={isSelected} />
+    </TabWrap>
+  );
+}
+
+export const Tab = forwardRef(BaseTab);
+
+const TabWrap = styled('li')<{
+  disabled: boolean;
+  orientation: Orientation;
+  overflowing: boolean;
+  selected: boolean;
+}>`
+  display: flex;
+  align-items: center;
+  position: relative;
+  height: calc(
+    ${p => p.theme.form.sm.height}px +
+      ${p => (p.orientation === 'horizontal' ? space(0.75) : 0)}
+  );
+  border-radius: ${p => p.theme.borderRadius};
+  transform: translateY(1px);
+
+  ${p =>
+    p.orientation === 'horizontal'
+      ? `
+        /* Extra padding + negative margin trick, to expand click area */
+        padding: ${space(0.75)} ${space(1)} ${space(1.5)};
+        margin-left: -${space(1)};
+        margin-right: -${space(1)};
+      `
+      : `padding: ${space(0.75)} ${space(2)};`};
+
+  color: ${p => (p.selected ? p.theme.activeText : p.theme.textColor)};
+  white-space: nowrap;
+  cursor: pointer;
+
+  &:hover {
+    color: ${p => (p.selected ? p.theme.activeText : p.theme.headingColor)};
+  }
+
+  &:focus {
+    outline: none;
+  }
+
+  ${p =>
+    p.disabled &&
+    `
+      &, &:hover {
+        color: ${p.theme.subText};
+        pointer-events: none;
+      }
+    `}
+
+  ${p => p.overflowing && `opacity: 0;`}
+`;
+
+const HoverLayer = styled('div')<{orientation: Orientation}>`
+  position: absolute;
```

Comment:
Why does this have to be a diff layer?

Reply:
(this applies to the focus layer too)
<img width="76" alt="Screen Shot 2022-09-19 at 2 53 21 PM" src="https://user-images.githubusercontent.com/44172267/191126343-0d5978e2-b1c3-473f-b196-05cd91c0c3ce.png">

The contents of outer wrap (`TabWrap`) extends all the way to `TabList`'s bottom border (the blue shaded area above). If we applied hover/focus effects to `TabWrap`, then the bounding box of those items will also extend to the bottom. I prefer a UI with a small gap at the bottom, making tab items look like little floating buttons:
<img width="74" alt="Screen Shot 2022-09-19 at 3 00 05 PM" src="https://user-images.githubusercontent.com/44172267/191127206-49260b2c-1274-438a-8d4b-69355a2c1dee.png">

That means we need separate hover/focus layers whose bounding boxes are smaller (shorter) than `TabWrap`'s bounding box.